src/routes/+page.svelte

```diff
@@ -51,14 +45,34 @@
 			loading = false;
 		}
 	}
+
+	async function createConversationWithMsg(message: string) {
+		// Ugly hack to use a store as temp storage, feel free to improve ^^
+		pendingMessage.set({
+			content: message,
+			files,
+		});
+
+		await createConversation();
+	}
+
+	async function createConversationWithPdf(pdfFile: File) {
+		pendingMessage.set({
+			files,
+			pdfFile,
+		});
+
+		await createConversation();
+	}
 </script>
 
 <svelte:head>
 	<title>{PUBLIC_APP_NAME}</title>
 </svelte:head>
 
 <ChatWindow
-	on:message={(ev) => createConversation(ev.detail)}
+	on:message={(ev) => createConversationWithMsg(ev.detail)}
+	on:uploadpdf={(ev) => createConversationWithPdf(ev.detail)}
```

Comment:
I'm not sure why we create an empty chat with a document, seems to me like it should be done like for images, where the files are "stored" in the front-end and the conversation created along when the first message is sent ?